CMake/kwiver-setup-python.cmake

```diff
@@ -204,6 +204,126 @@ mark_as_advanced(PYTHON_ABIFLAGS)
   find_package(pybind11)
 
 
+###
+# Python Dependencies
+#
+# sets the python dependencies defined in python/requirements.txt
+# to be a custom command of the python libraries target
+# a venv will be created to encapsulate the pip installed dependencies
+# from the larger system, while still providing the tests access to their dependencies
+#
+if (KWIVER_ENABLE_TESTS)
+
+  message(STATUS "Python and testing enabled.")
+  message(STATUS "Searching for Python3 Interp...")
+  # locate the python3 install to gather info for venv creation
+  # determine if the package is conda
+  find_package(Python3 COMPONENTS Interpreter)
+  set(CREATE_VENV 1)
+  if (Python3_INTERPRETER_ID STREQUAL "Anaconda")
+    # conda venvs are managed by conda package manager
+    # require conda specific command
+    message(STATUS "Python found is: Anaconda")
+    set(CONDA 1)
+    set(VENV_DIR "testing_venv")
+    set(CREATE_VENV_ARG create -n ${VENV_DIR} pip --yes)
+    if(DEFINED ENV{CONDA_EXE})
+      set(CREATE_VENV_CMD $ENV{CONDA_EXE})
+    else()
+      message(WARNING "Could not find conda executable, conda env will NOT be created."
+                      "Python tests may not be run.")
+      unset(CREATE_VENV)
+    endif()
+  else()
+    set(VENV_DIR "${KWIVER_BINARY_DIR}/testing_venv")
+    set(CREATE_VENV_CMD ${Python3_EXECUTABLE})
+    set(CREATE_VENV_ARG -m venv "${VENV_DIR}" )
+  endif()
+  if (CREATE_VENV)
+    message(STATUS "Creating virtualenv @${VENV_DIR} for testing...")
+    execute_process (
+                      COMMAND ${CREATE_VENV_CMD} ${CREATE_VENV_ARG}
+                      RESULT_VARIABLE create_venv_result
+                      COMMAND_ECHO STDOUT
+                    )
+    if (create_venv_result AND NOT create_venv_result EQUAL 0)
+        # could not create venv, report to that effect, Nose may still be found and tests may still be run
+        # but dependencies (including nose) to be met by a pip install are not garunteed
+        message (WARNING "Virtualenv creation failed. Python tests may not be run or may fail unexpectedly.")
+    else()
+      set(VENV_CREATED 1)
+    endif()
+    message(STATUS "Virtualenv creation exited with status: ${create_venv_result}")
+    unset(Python3_FOUND)
+    if (VENV_CREATED)
+      if (APPLE)
+        set(CMAKE_FIND_FRAMEWORK "NEVER")
+      elseif (WIN32)
+        set(Python3_FIND_REGISTRY "NEVER")
+      endif()
+      include("${KWIVER_SOURCE_DIR}/CMake/utils/kwiver-config-venv.cmake")
```

Comment:
Pretty minor, but you could probably use ${KWIVER_CMAKE_DIR}/utils here. That variable is defined in the top level CMakeLists.txt so should be available here. Small gains for future-proofing that location.